---
source: tests/common/mod.rs
description: "// This is a regression test. There was a bug where if an upvalue for an\n// earlier local (here \"a\") was captured *after* a later one (\"b\"), then it\n// would crash because it walked to the end of the upvalue list (correct), but\n// then didn't handle not finding the variable.\n\nfun f() {\n  var a = \"a\";\n  var b = \"b\";\n  fun g() {\n    print b; // expect: b\n    print a; // expect: a\n  }\n  g();\n}\nf();\n"
input_file: crafting_interpreters_test_files/closure/close_over_later_variable.lox
---
OUT: 
b
a

===
ERR:
